### ISSUE TITLE
Handle new-initiator/new-responder correctly

### ISSUE THREAD

**Issue by lgrahl** [saltyrtc/saltyrtc-client-rs#61]
When a 'new-initiator' or 'new-responder' message has been received, the initiator/responder simply replaces the corresponding instance but does not reset its own state.

Due to a limitation of the implementation, after the task has kicked in, it should also...

- drop the responder in case it is an initiator, and
- close the connection normally in case it is a responder.

**Comment by dbrgn** [comment]
Good point. I assume with "its own state" you mean some fields in the `Common` struct.

If I'm not mistaken, if the task has not yet kicked in, we only need to reset `self.common.signaling_state`.
